core/app/models/spree/country.rb

```diff
@@ -13,9 +13,12 @@ def self.states_required_by_country_id
     end
 
     def self.default
-      if default_country_id = Spree::Config[:default_country_id]
-        find_by_id(default_country_id)
-      end || first
+      if Spree::Config.default_country_id
+        ActiveSupport::Deprecation.warn("Setting your default country via its ID is deprecated. Please set your default country via the `default_country_iso` setting.", caller)
+        find_by(id: Spree::Config.default_country_id) || find_by!(iso: Spree::Config.default_country_iso)
```

Comment:
I don't understand why you want to load the country by `iso` if the find by id returns `nil`